### DISCUSSION TITLE
skywalking go agent error with sw8 header

### DISCUSSION THREAD

**Discussion by xiaozaiyuji** [apache/skywalking#11229] [Q&A]
![image](https://github.com/apache/skywalking/assets/4230320/3be87394-b0a8-4e5c-8f6d-f238ddcd0771)
I find the header become big than big as an error config by  go service use http client to transfer  self http interface.
To my surprise as I print the header ,I find the sw8 header show as many as the interface transfer.

**Comment by wu-sheng** [comment]
I think the HTTP request is enhanced by multiple plugins duplicated.

FYI @mrproliu

**Reply by mrproliu** [reply]
Yes, It could be enhanced by multiple plugins. But if they are in the same gorouine, the header should not be re-added, so, it's related to the frameworks.

**Reply by wu-sheng** [reply]
@xiaozaiyuji You are better to provide a demo app to help reproduce the case.

**Comment by mrproliu** [comment]
Could you please provide more context for it? such as which agent version, or framework is used.

**Reply by xiaozaiyuji** [reply]
0.1.0 gin

**Reply by xiaozaiyuji** [reply]
If the agent 0.2.0 solve this problem I will close this discuss . I will reply soon as I did this test with 0.2.0.

**Reply by mrproliu** [reply]
Only gin server? which framework are you using for sending request? If you could provide a simple demo program, that would be great.
